### DISCUSSION TITLE
如何给局域网内不同的v6地址绑定不同域名

### DISCUSSION THREAD

**Discussion by threerog** [jeessy2/ddns-go#648] [Q&A]
目前我是运行在openwrt上并且成功的将路由器192.168.5.1绑定域名(a.example.com)+外网访问
但是我路由器和局域网内还有几个服务
`192.168.5.1:8888`
`192.168.5.10:9999`
我目前是通过端口转发的方式来进行访问
`a.example.com:8888`
`a.example.com:9999`

能不能实现
`192.168.5.1:8888 => b.example.com`
`192.168.5.10:9999 => c.example.com`

**Comment by jeessy2** [comment]
其它机器安装ddns-go
或 使用`通过命令获取` IPv6前缀+EUI64

**Reply by threerog** [reply]
> 其它机器安装ddns-go 或 使用`通过命令获取` IPv6前缀+EUI64

第二种方式能不能详细讲讲 实操一下

**Reply by jeessy2** [reply]
第二种，要其他设备支持eui64的IPv6，包含ff:ee
剩下的自己探索，不复杂

**Reply by shichunjian** [reply]
如果我只想在路由器上获取内网设备的ipv6地址，并解析到指定的域名，而且不是在不同的设备上安装ddns-go，这个怎么做，DDNS服务可以通过增加新的服务做到，ddns-go真的不知道怎么做，请指教。

**Comment by threerog** [comment]


**Reply by jeessy2** [reply]
域名+端口不就是？

**Comment by threerog** [comment]
我最上面提的那个问题 假设设备支持你说的eui64的IPv6 具体该如何操作
我看了下discussions和issues里不止我一人在咨询这个问题
ddnsgo能否对此类需求适配相应的UI配置以及配置教程，我是刚入门的小白
我可以提供一定的开源赞助费用(ps:好像没有找到项目的开源赞助信息)

**Reply by jeessy2** [reply]
很简单的，我就是没空来写文档，如果那位有意愿可写一个

**Reply by WaterLemons2k** [reply]
根据 MAC 地址生成 IPv6 地址：
- [sed - How to use shell to derive an IPv6 address from a MAC address? - Unix &amp; Linux Stack Exchange](https://unix.stackexchange.com/a/557208)
- [AndreBL/ip6neigh: Giving local DNS names to IPv6 SLAAC addresses [OpenWrt/LEDE shell script]](https://github.com/AndreBL/ip6neigh)
- [MAC address to IPv6 link-local address online converter &#8211; Ben&#039;s blog](https://ben.akrin.com/mac-address-to-ipv6-link-local-address-online-converter/)

**Comment by threerog** [comment]
以下问题不属于`ddns-go`的范畴，仅交流
如何给同一ipv6地址的不同端口绑定不同域名的相同端口且互不干扰
通过`ddns-go`将`a.example.com`和`b.example.com`解析到了ipv6对应设备
实现如下情况
`a.example.com=>[::]:8888`
`b.example.com=>[::]:9999`
注意这里的域名都使用默认的HTTP80端口


上面这种情况有没有大佬提供解决思路

**Reply by michaelauto** [reply]
反向代理…
